Database/MongoDB/Query.hs

```diff
@@ -1272,9 +1272,9 @@ find q@Query{selection, batchSize} = do
         qr <- queryRequestOpMsg False q
         let newQr =
               case fst qr of
-                Req qry ->
-                  let coll = last $ T.splitOn "." (qFullCollection qry)
-                  in (Req $ qry {qSelector = merge (qSelector qry) [ "find" =: coll ]}, snd qr)
+                Req P.Query{..} ->
```

Comment:
Is this hunk necessary? Seems to be doing the same as before? Probably a remnant from experimenting with the code?

Reply:
This gets rid of the use of a partial record selector function (which would produce a warning). I think it's a good practice to avoid using record selector functions to sum types

Reply:
I see. It would be better to have refactorings in a separate commit. But it's good for now.